src/ray/gcs/gcs_client/gcs_client.cc

```diff
@@ -177,7 +146,44 @@ std::pair<std::string, int> GcsClient::GetGcsServerAddress() const {
 
 PythonGcsClient::PythonGcsClient(const GcsClientOptions &options) : options_(options) {}
 
-Status PythonGcsClient::Connect() {
+namespace {
+Status HandleGcsError(rpc::GcsStatus status) {
+  RAY_CHECK(status.code() != static_cast<int>(StatusCode::OK));
+  return Status::Invalid(status.message() +
+                         " [GCS status code: " + std::to_string(status.code()) + "]");
+}
+}  // namespace
+
+Status PythonGcsClient::Connect(ClusterID &cluster_id, int64_t timeout_ms) {
```

Comment:
The reference of cluster_id makes it an in and out parameter is confusing.

Maybe let's add const there. Keep a variable inside the class as cluster_id. Give a getter function.